test/test_torch.py

```diff
@@ -15450,7 +15451,7 @@ def run_subtest(matrix_size, batches, device, pivot, singular=False, a=None):
                 else:
                     rows, columns = matrix_size
                 if a is None:
-                    a = random_matrix(rows, columns, *batches, **dict(singular=singular)).to(device)
+                    a = random_matrix(rows, columns, *batches, **dict(singular=singular, dtype=dtype)).to(device)
```

Comment:
looks like the dtype is always set to `torch.double` https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/common_utils.py#L1571

Reply:
It a value by default.